---
title: "project-description"
output: md_document
date: "`r Sys.Date()`"
---

## Introduction into the topic

Salvador et al. (2023) published a research article on age-dependent immune and lymphatic responses following spinal cord injury to better understand the lifelong disabling effects that can result from such severe injuries [Publication](https://www.cell.com/neuron/fulltext/S0896-6273(23)00296-9?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0896627323002969%3Fshowall%3Dtrue#mmc2).


The study investigates how the immune and lymphatic systems respond to spinal cord injuries in young and old mice. The researchers used a technique called single-cell RNA sequencing to study the activity of individual cells by measuring which genes are turned on in each cell (scRNA-seq).

## Data

The data is publicly available on the Gene Expression Omnibus (GEO) under the accession number GSE224123. The pre-analyzed data set mmc4.xlsx is already provided on GitHub [Data file](https://github.com/Dr-Eberle-Zentrum/Data-projects-with-R-and-GitHub/blob/4e680979e0d3a5a57912ffd151ef1e9acca38c20/Projects/lizzola/mmc4.xlsx).
To give you an idea how the data looks like:

```{r table, echo=FALSE, results = 'asis', message=FALSE}

library(tidyverse)
library(knitr)
library(dplyr)

data <- tibble(
  gene = c("Lars2"),
  logFC = c(1.705495961),
  p_adjust = c(1.59E-35),
  ID = c("GO:0045766"),
  Description = c("positive regulation of angiogenesis"),
  GeneRatio = c("11/109"),
  BgRatio = c("188/23328"),
  pvalue = c(1.34E-09),
  p_adjust_2 = c(3.21E-06)
)

kable(data)

```

### Data manipulation

- The excel sheet contains multiple sheets, so the first task is to get familiar with the data file. Find the sheet "Aged vs Young Macrophages". 
- Extract the sheet "Aged vs Young Macrophages". It contains the differentially expressed genes (DEGs) and their associated GO terms for different time points (naïve, 3dpi, 7dpi, 14dpi). We are only interested in the naïve aged macrophages vs. naïve young macrophages comparison.
- Create a new column called logFC_high that is TRUE if logFC > 2 for a gene of "naïve aged macrophages vs. naïve young macrophages" , and FALSE otherwise.
- Filter the DEGs to include only those with p.adjust < 0.05, and return the top 10 genes with the highest logFC. Please display the top 10 genes in a nice table.
- Filter the GO terms to include only those whose description contains keywords such as "angiogenesis", "immune response", or "cytokine".


### Data visualization

- Create a bar plot for the top 10 genes with logFC > 2 and p.adjust < 0.05. Put the genes on the y-axis. Label the axes accordingly and add a title.

- Color the bars based on the logFC_high status (TRUE or FALSE).

- Assign each gene a corresponding GO term (e.g., from a previous analysis) and display it as an additional label on the plot (e.g., as text above or beside the bars). Group the genes by their corresponding GO term and use different colors for each group.