---
title: "RNAseq in pregnancy - Solution lizzola"
output: md_document
date: "`r Sys.Date()`"
---

## Data manipulation

Import the data

```{r data set, echo=TRUE, message=FALSE, warning=FALSE}
library(dplyr)
library(tidyr)
library(tidyverse)
library(kableExtra)
library(knitr)
library(fs)
library(pheatmap)


if (!is.null(current_input())) {
  rmd_dir <- path_dir(current_input())
} else {
  rmd_dir <- getwd()
}
data <- read.csv(file.path(rmd_dir, "GSE273780_EukmRNAseq_counts.csv"), row.names = 1)
head(data)
```

Filter out the count data of the following columns (G25, G26, G27, G30, G9, G12, G32, G45). Rename the columns.

| Sample | Week   | Type | Replicate |
|--------|--------|------|-----------|
| G25    | Week12 | NW   | 1         |
| G26    | Week12 | NW   | 2         |
| G27    | Week12 | OW   | 1         |
| G30    | Week12 | OW   | 2         |
| G9     | Week36 | NW   | 1         |
| G12    | Week36 | NW   | 2         |
| G32    | Week36 | OW   | 1         |
| G45    | Week36 | OW   | 2         |

```{r data, echo = TRUE}

filtered_data <- data %>%
  select(Row.names, G25, G26, G27, G30, G9, G12, G32, G45) %>%
  rename(Gene = Row.names, 
         Week12_NW_1 = G25, 
         Week12_OW_1 = G26, 
         Week12_NW_2 = G27, 
         Week12_OW_2 = G30, 
         Week36_NW_1 = G9, 
         Week36_OW_1 = G12, 
         Week36_NW_2 = G32, 
         Week36_OW_2 = G45)

long_data <- filtered_data %>%
  pivot_longer(-Gene, names_to = "Condition", values_to = "Count") %>%
  separate(Condition, into = c("Week", "Type", "Replicate"), sep = "_")

```

Normalize the data from counts to counts per million (add up all the counts per sample and divide each sample by this, then multiply by 1.000.000). log2 transformation of the counts per million to make the data more symmetric.

```{r transformation, echo = TRUE}

log2_data <- long_data %>%
  group_by(Week, Type, Replicate) %>%
  mutate(
    Total = sum(Count, na.rm = TRUE),
    CPM = Count / Total * 1e6,
    CPM_log2 = log2(CPM + 1)
  ) %>%
  ungroup()

```

Drop any rows with blank names and count values between 0 and 2.

```{r Filter, echo = TRUE}
log2_data_under_2 <- log2_data %>%
  filter(!is.na(Gene) & Gene != "") %>%
  rowwise() %>%
  filter(CPM_log2 >= 2) %>%
  ungroup()

```

Calculate the z-Value of each count in new columns by subtracting the normalized count value by the mean of the column and divide by the columns standard deviation. Calculate the variance of the log2 transformed counts by rows and put it in a new column.

```{r z-value and variance, echo = TRUE}

heatmap_data <- log2_data_under_2 %>%
  group_by(Week, Type, Replicate) %>%
  mutate(CPM_z = (CPM - mean(CPM, na.rm = TRUE)) / sd(CPM, na.rm = TRUE)) %>%
  ungroup() %>%
  group_by(Gene) %>%
  mutate(log2_CPM_variance = var(CPM_log2, na.rm = TRUE)) %>%
  ungroup()

```


## Data visualization

Rank all genes descendingly by variance. Keep only top 50 highest-variance genes for the heat map.
```{r heatmap plot, echo = TRUE}

top50_genes <- heatmap_data %>%
  distinct(Gene, log2_CPM_variance) %>%
  slice_max(log2_CPM_variance, n = 50) %>%
  pull(Gene)

heatmap_matrix <- heatmap_data %>%
  filter(Gene %in% top50_genes) %>%
  select(Gene, Week, Type, Replicate, CPM_log2) %>%
  unite("Sample", Week, Type, Replicate, sep = "_") %>%
  pivot_wider(names_from = Sample, values_from = CPM_log2) %>%
  column_to_rownames("Gene") %>%
  as.matrix()

heatmap_matrix <- heatmap_matrix[, c(
  "Week12_NW_1", "Week12_NW_2", "Week12_OW_1", "Week12_OW_2",
  "Week36_NW_1", "Week36_NW_2", "Week36_OW_1", "Week36_OW_2"
)]

colnames(heatmap_matrix) <- gsub("_", " ", colnames(heatmap_matrix))

gaps <- seq(2, ncol(heatmap_matrix) - 1, by = 2)

annotation_colors <- list(
  Week = c("Week12" = "coral", "Week36" = "lightgreen"),
  Type = c("NW" = "lightgoldenrod", "OW" = "lightyellow")
)

heatmap_matrix[is.na(heatmap_matrix)] <- 0

png("heatmap.png", width = 1200, height = 1200)
pheatmap(
  heatmap_matrix,
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  annotation_colors = annotation_colors,
  annotation_col = data.frame(
    Week = factor(rep(c("Week12", "Week36"), each = 4)),
    Type = factor(rep(c("NW", "OW"), times = 4)),
    row.names = colnames(heatmap_matrix)
  ),
  scale = "column",
  color = colorRampPalette(c("royalblue1", "white", "pink3"))(100),
  main = "Heatmap of Top 50 Genes by Variance",
  fontsize_row = 5,
  fontsize_col = 10,
  angle_col = 45,
  border_color = "grey70",
  gaps_col = gaps
)

dev.off()

```
![Heatmap Top 50 Genes](heatmap.png)

#### Conclusion

The heatmap shows that gene expression of individual genes differs between the two weight groups (Normal Weight and Overweight). The variability within the replicates is also interesting. The expression pattern change between weeks (Week 12 and Week 36) indicates that the gene expression landscape is dynamic during pregnancy, although at first glance the changes in gene expression between the two weight groups appear to be similar.
To conclude, gene expression is influenced by both gestational age and maternal weight status.