---
title: Analysing RNAseq data of immun cell associated with gut microbiom of normal
  weight and over weight woman during pregnancy
output: md_document
---
## Background of the dataset
This data was acquired from two groups of healthy pregnant woman one with normal weight (NW, BMI 18.5-24.9) and one slightly overweight (OW, BMI 25-29.9). the researchers looked at the mothers immune cells to find out how the difference in BMI affect those cells during pregnancy and how those changes can link back to the diet, microbiome or baby microbiome.

Source of the Data:
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE273780 

[RNA_seq_data.csv](https://github.com/Dr-Eberle-Zentrum/Data-projects-with-R-and-GitHub/blob/main/Projects/t1mge/GSE273780_EukmRNAseq_counts.csv)

The columns of the data set each correspond to one sample, so G92 could be pregnant woman at week 36 normal weight rep 2 or similar. For later we are only interested in some of these observations. 

To find out which conditions correspond to which columns go to the source of the data and go to the "Samples (59)" section there you can find all conditions and by clicking, you get redirectd to a new window were you can find the corresponding column names under the Description tab. For example pregnant woman at week 12 normal weight rep 1 is G25.

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.width=12, fig.height=10}
rna_seq <- read.csv(
  "https://raw.githubusercontent.com/Dr-Eberle-Zentrum/Data-projects-with-R-and-GitHub/refs/heads/main/Projects/t1mge/GSE273780_EukmRNAseq_counts.csv",
  row.names = NULL,
  check.names = TRUE
)
head(rna_seq, n = 10)
```


## Wrangling goals
1. Import the data
2. Figure out what you are looking 
3. Filter out the count data of 2 replicates each of normal weight and overweight woman at week12 and at week 36 (columns are the samples and rows are the specific genes) so that you have the rows left for week12(Normal‑1, Normal‑2, Overweight‑1, Overweight‑2)  and for week36 the same ones 
4. Keep the column, external gene names 
5. rename the columns with more readable names
6. Normalize the data from counts to counts per million (add up all the counts per sample and divide each sample by this, then multiply by 1.000.000)
7. log2 transformation of the counts per million to make the data more symmetric
8. Drop any rows with blank names and count values between 0 and 2
9. Calculate the z-Value of each count in new columns by subtracting the normalized count value by the mean of the column and divide by the columns standard deviation
10. Calculate the variance of the log2 transformed counts by rows and put it in a new column.

With these steps you should end-up with the z-values and the varaince which are used in the heatmap



## Visualization Goals
### Produce a clustered heatmap of the top50 genes 


1. Rank all genes  descendingly by variance
2. keep only top 50 highest-variance genes for the heat map
3. Rows should be the external gene names; columns the samples
4. produce a heat map looking something like blow.

You can use this resource to get more information on how to create heatmaps: https://r-graph-gallery.com/heatmap

If you get a simple heatmap with just the annotations, that would already be amazing.

Further if you have a lot of time you can create an interactive heatmap. In which you can hover over the tiles and get information on the z-value, the gene name and the sample description. An example is also shown on the resource for heatmaps  form above. 
 
I hope it is going to work out for you!
 
Good Luck!

![](https://raw.githubusercontent.com/Dr-Eberle-Zentrum/Data-projects-with-R-and-GitHub/refs/heads/main/Projects/t1mge/Heatmap%20beispiel.png)